### PR TITLE
Feature/ch19186/add shared link to shared with me

### PR DESCRIPTION
**Pull request by perfectmak**
**CHANGELOG**
- Update OpenSharedFile logic to add successfully opened file to ReceivedFiles thread
- Also, allows re-opening shared files without requiring password all the time
- Update the `SharedListDirectoryEntry` message proto schema to include `isPublicLink` to indicate the shared file is a public link item

### CODE REVIEW

File: core/textile/sharing.go
Comment (dmerrill6) [inline core/textile/sharing.go:200]: Will this offset work even when we are merging two entity lists?

File: grpc/proto/space.proto
Comment (dmerrill6) [inline grpc/proto/space.proto:368]: Don't we also have to add the `password` field to the proto?

**Review by dmerrill6 — APPROVED** [review]
LGTM, left two questions

File: grpc/proto/space.proto
Comment (perfectmak) [reply grpc/proto/space.proto:368]: No, it is not useful. After the first time the open public file rpc is called, because that logic persists the password and automatically refetches it on subsequent calls.

So the frontend would pass an empty password when opening public link files from the shard with me response

File: core/textile/sharing.go
Comment (perfectmak) [reply core/textile/sharing.go:200]: Yeah it should still work. We are not really merging two lists, it is only a single list being queried, but depending on the type of it's data we map it to a public shared file or a direct invite shared file.

File: grpc/proto/space.proto
Comment (dmerrill6) [reply grpc/proto/space.proto:368]: How do they pass the password then?

File: core/textile/sharing.go
Comment (dmerrill6) [reply core/textile/sharing.go:200]: Understood 👍

File: grpc/proto/space.proto
Comment (perfectmak) [reply grpc/proto/space.proto:368]: They pass it only on the `OpenPublicFile` rpc request

Comment (dmerrill6) [reply grpc/proto/space.proto:368]: I don't see a field for password there

Comment (dmerrill6) [reply grpc/proto/space.proto:368]: I don't see a field for password there

File: core/space/services/services_sharing.go
Comment (jsonsivar) [inline core/space/services/services_sharing.go:221]: Maybe we update the name to OpenPublicSharedFile?

File: core/textile/model/model.go
Comment (jsonsivar) [inline core/textile/model/model.go:44]: Noticed we are passing in the pw here so maybe we can update the name (technically doesn't make a difference since they are both strings but just in case).

**Review by jsonsivar — APPROVED** [review]
LGTM once merge conflict is resolved and I think as per @dmerrill6 's comment, we need to add the password field for `OpenPublicFile` before merging.

**Pull request merged**
